lib/build.js

```diff
@@ -209,6 +210,34 @@ class BuildModel extends BaseModel {
     }
 
     /**
+     * Abort running steps. If no steps ever ran, abort the first step
+     * @method abortSteps
+     */
+    abortSteps() {
+        const now = (new Date()).toISOString();
+        let steps = this.steps;
+
+        // fail first step if never started
+        if (!steps[0].startTime) {
+            steps[0].startTime = now;
+            steps[0].endTime = now;
+            steps[0].code = ABORT_CODE;
```

Comment:
I made that assumption at first, but we've clearly seen cases where it looks like the build failed when no steps ever started.
The log api doesn't care what start/end times are. It only looks for an exit code to determine if the step is finished, so every step needs to have an exit code in order to prevent the log api from returning x-more-data=true. While the UI is nice and doesn't allow you to open logs views for steps that don't appear to have started, nothing prevents someone from asking the api directly.